pkg/kubeletclient/kubeletclient.go

```diff
@@ -18,27 +17,26 @@ import (
 const (
 	defaultKubeletSocketFile   = "kubelet.sock"
 	defaultPodResourcesMaxSize = 1024 * 1024 * 16 // 16 Mb
-)
-
-var (
-	kubeletSocket           string
 	defaultPodResourcesPath = "/var/lib/kubelet/pod-resources"
 )
 
 // GetResourceClient returns an instance of ResourceClient interface initialized with Pod resource information
-func GetResourceClient() (types.ResourceClient, error) {
+func GetResourceClient(kubeletSocket string) (types.ResourceClient, error) {
 	// If Kubelet resource API endpoint exist use that by default
 	// Or else fallback with checkpoint file
-	if hasKubeletAPIEndpoint() {
+	if kubeletSocket == "" {
+		kubeletSocket = util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
```

Comment:
this returns a string with the scheme of the endpoint i.e
"unix:/pkg/kubelet.sock"
https://play.golang.org/p/zqf-bAWCnVt
does os.Stat() inside `hasKubeletAPIEndpoint()` detects it?

Reply:
oh, good catch! Thank you for your comment! will fix it.